### ISSUE TITLE
captureRef not taking full picture of scrollview

### ISSUE THREAD

**Issue by cheehau** [gre/react-native-view-shot#274]
Hi

I am trying to use captureRef to take shot of scrollview that extends beyond the screen. the width and height of the final image seems to take after the scrollview which is correct. But the content stop at the screen height. after the screen height it is just a dark patch.

Anyone have this issue?

Thanks

**Comment by DKKUKADIYA** [comment]
Yes, I have same issue
Please anyone share proper solution

**Comment by gre** [comment]
@cheehau @DKKUKADIYA 
- have you tried to use `snapshotContentContainer` ?
- also, you can try to make sure to capture the inner part of your scrollview, not the scrollview itself. like shown in https://github.com/gre/react-native-view-shot#recommended-high-level-api
- do you have a snippet of code we can look at to investigate?

**Comment by kraison1** [comment]
You can try my method.

<ViewShot
            ref="viewShot"
            style={{flex: 1, backgroundColor: '#FFFFFF'}}>
            
</ViewShot>

**Comment by outaTiME** [comment]
Im running into the same issue only on android with expo 36 and version 3.0.2 of react-native-view-shot, here my snippet:

```js
import { captureRef } from 'react-native-view-shot';
import * as Sharing from 'expo-sharing';

// action

captureRef(this._shareViewContainer, {
  snapshotContentContainer: true,
}).then(
  uri => {
    console.log('Snapshot uri', uri);
    // https://github.com/expo/expo/issues/6920#issuecomment-580966657
    Sharing.shareAsync('file://' + uri);
  },
  error => console.error('Oops, snapshot failed', error)
);

// render

<ScrollView ref={component => this._shareViewContainer = component}>
  <View
    style={{
      flex: 1,
      padding: 16,
    }}>
      // items
  </View>
</ScrollView>
```js

Here the complete screen on Android: 

![telegram-cloud-photo-size-1-5028670258914896071-y](https://user-images.githubusercontent.com/140472/73775314-6aa94000-4764-11ea-8635-635c83b83466.jpg)

And here the `react-native-view-shot` output on Android : 

![telegram-cloud-photo-size-1-5028670258914896070-y](https://user-images.githubusercontent.com/140472/73775231-48afbd80-4764-11ea-975a-6230e81595c8.jpg)

But on iOS, here the complete screen:

![IMG_0003](https://user-images.githubusercontent.com/140472/73775609-00dd6600-4765-11ea-9168-9272a6d4e8b7.PNG)

And here the output:

<img width="320" alt="91682F3E-D1F8-42B3-B166-5BA767EE9CDF" src="https://user-images.githubusercontent.com/140472/73775622-08047400-4765-11ea-9e45-9f1c58507777.png">

in short, it seems that in android, the hidden part of scroll is not exported on image, any advice will be welcome, 

Thanks !!!

**Comment by outaTiME** [comment]
By the moment i solve the issue wraping `ScrollView` items contents inside `View, here the final snippet:

```js
import { captureRef } from 'react-native-view-shot';
import * as Sharing from 'expo-sharing';

// action

captureRef(this._shareViewContainer, {
  // snapshotContentContainer: true,
}).then(
  uri => {
    console.log('Snapshot uri', uri);
    // https://github.com/expo/expo/issues/6920#issuecomment-580966657
    Sharing.shareAsync('file://' + uri);
  },
  error => console.error('Oops, snapshot failed', error)
);

// render

const FOOTER_HEIGHT = 21;
const PADDING = 16;
const BOTTOM_MARGIN_FOR_WATERMARK = FOOTER_HEIGHT * PADDING;

<ScrollView 
  style={{
    marginBottom: -(BOTTOM_MARGIN_FOR_WATERMARK),
  }}
  contentContainerStyle={{
    flexGrow: 1,
  }}
  scrollIndicatorInsets={{
    bottom: BOTTOM_MARGIN_FOR_WATERMARK,
  }}>
  <View 
    style={{ 
      flex: 1, 
      borderRadius: PADDING / 2, 
    }}
    ref={component => this._shareViewContainer = component}>
    <View
      style={{
        flex: 1,
        padding: PADDING,
      }}>
      // items
    </View>
    <View style={{ height: FOOTER_HEIGHT }}>
      // footer
    </View>
  </View>
</ScrollView>
```

I hope it helps,

**Comment by Gabriel637** [comment]
React dont accept refs in components, but when removed, I receive this error
![Simulator Screen Shot - iPhone 11 - 2020-02-14 at 18 26 02](https://user-images.githubusercontent.com/37639798/74568764-79a7a380-4f57-11ea-993b-aba24766c8d0.png)

**Comment by mutaimwiti** [comment]
Have a `View` inside the `ScrollView` and capture it instead of the `ScrollView`.

```
<ScrollView style={{flex: 1}}>
  <View
    collapsable={false}
    ref={view => this.statsRef = view}
  >
    <Stats stats={stats}/>
  </View>
</ScrollView>
```

**Comment by LuizFBrisighello** [comment]
> Have a `View` inside the `ScrollView` and capture it instead of the `ScrollView`.
> 
> ```
> <ScrollView style={{flex: 1}}>
>   <View
>     collapsable={false}
>     ref={view => this.statsRef = view}
>   >
>     <Performance data={data}/>
>   </View>
> </ScrollView>
> ```

This worked for me

**Comment by FazilMuhammed** [comment]
@mutaimwiti can you share full code ?

**Comment by JinleeJeong** [comment]
> Have a `View` inside the `ScrollView` and capture it instead of the `ScrollView`.
> 
> ```
> <ScrollView style={{flex: 1}}>
>   <View
>     collapsable={false}
>     ref={view => this.statsRef = view}
>   >
>     <Performance data={data}/>
>   </View>
> </ScrollView>
> ```

@FazilMuhammed It is full code. You can use a component instead of the `Performance` component.
This worked for me! Thank you :)

**Comment by pallaviMN** [comment]
@kraison1 , can u please help me in ths? I too used Viewshot instead of view but it captures only visible region. Is it possible to send full code?

**Comment by gre** [comment]
as I believe there are solution presented here, I will close the issue (comment can still happen on closed tickets)

**Comment by yaser-elbatal** [comment]
> 

 when i pass ref to View 
 the App crash
any help

**Comment by vinodsantharam** [comment]
I struggled to make it work. 
So the `ref` has to be set to the `ScrollView`.

```
 <ScrollView ref={viewShotRef}>
            <ViewShot>
                     <YourComponent />
            </ViewShot>
 </ScrollView>
```

I have a button on the same screen. When the user presses it, the callback calls `captureRef`. Ensures that you have the options object with `snapshotContentContainer: true`.

```
const share = useCallback(async () => {
    if (viewShotRef.current) {
      const snapshot = await captureRef(viewShotRef, {
        result: 'data-uri',
        snapshotContentContainer: true,
      });
      // DO WHATEVER YOU WANT WITH THE SNAPSHOT
      }
    }
  }, [viewShotRef]);
```

**Comment by FabioRocha462** [comment]
Hi, can anyone help me? 
I have an image that I need to leave only the rounded part, and remove this gray art, how can I do that?
<img width="360" alt="3BD73951-FDCB-4A16-94BD-E3445544DF66" src="https://github.com/gre/react-native-view-shot/assets/86632788/e80a825c-32cc-4316-b54d-d9eff83ebfc2">
